src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -658,6 +657,12 @@ public void init() {
             mainStage.titleProperty().bind(windowTitle);
         });
         initShowTrackingNotification();
+        booleanBinding = Bindings.isEmpty(tabbedPane.getTabs());
+        // Adding the welcome
+        welcomePage = WelcomePage.getInstance(prefs, stateManager, fileUpdateMonitor, taskExecutor, dialogService, welcomeBox);
```

Comment:
Why a singleton? Why so much drama? ;-) Just a simple pane or box with some stuff as children should be sufficient.